### DISCUSSION TITLE
Running Banking example on XVM getting build failed

### DISCUSSION THREAD

**Discussion by Boardmad** [xtclang/xvm#149] [Q&A]
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':server:compileAppModule'.
> Process 'command 'xcc'' finished with non-zero exit value 255

From stacktrace: 

Failed to generate code for Compiler (Module=BankStressTest, Stage=Emitting)
java.lang.IllegalStateException: Failure to make progress on a TypeInfo for AnnotatedType{type=@Atomic Ref<Int>}; deferred types=[AccessType{type=@Atomic Ref<Int>:private}]
Errors: Branched: Count=1, Severity=ERROR, Last=VERIFY-32: "Ref<Int>" is annotated by "Atomic<Atomic.Referent>", but is not compatible with the required "into": "Var<Atomic.Referent>".

I'm rather new to the community but keen to learn - guidance for stoopid appreciated?

**Comment by ggleyzer** [comment]
Thank you for your report!

This issue has been resolved by the change in xvm repository:

         [Fix a compiler assert Gene Gleyzer 2/19/24, 12:32 PM](https://github.com/xtclang/xvm/commit/b3e89b4997f21fcef47aec6af685a8da9276df70)

Also, just yesterday I pushed a big reworking to the platform repo, which may require a complete wipe out of the ~xqiz.it directory.

Please rebuild or reload the xdk, the platform and try again.

**Reply by Boardmad** [reply]
Thanks Gene
Wiped xqiz.it and reloaded xdk but rebuild of platform now:
Error: /Users/xtc/Development/platform/common/src/main/x/common/HostInjector.x:49 [49:19..49:36] COMPILER-38: Name "Unsupported" is unresolvable. ("new Unsupported()")

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'platform'.
> Could not create task ':build'.
   > Could not create task ':kernel:build'.
      > Could not create task ':common:build'.
         > Process 'command 'xcc'' finished with non-zero exit value 255

**Reply by Boardmad** [reply]
Morning Gene
Just pulled latest, wiped with a clean build and still getting same unsupported failure

ran brew reinstall xdk-latest
Warning: xtclang/xvm/xdk-latest 0.4.3.934 is already installed and up-to-date.

**Reply by ggleyzer** [reply]
You're right; it looks like our brew distribution is busted. As a work around, please build the xvm project and set XDK_HOME environment variable to "[your repo path]/xvm/xdk/build/install/xdk". 

Also, you'd need to copy the launcher (just once):

    cp [your repo path]/xvm/xdk/build/install/xdk/bin/macos_launcher /opt/homebrew/Cellar/xdk-latest/0.4.3.934/libexec/bin

**Comment by ggleyzer** [comment]
Hmm... That should have been fixed by the [last commit to the platform](https://github.com/xtclang/platform/commit/7618ee90501bb151ec6f7a33aaef85dd5b52604a#diff-8f851108fe335277ba5788f7ed547d9bd5b64504b5a8433165b56f5dc568759a)

**Reply by Boardmad** [reply]
Will pull again and rebuild in a bit and restest

**Reply by ggleyzer** [reply]
Sorry, got a bit confused - it appears you are not on the latest XDK version. Cam renamed `UnsupportedOperation` to `Unsupported` in [this change](https://github.com/xtclang/xvm/commit/d1515b6d7052cae092ba435e13afdce77df60cb0)

**Reply by Boardmad** [reply]
thanks and disregard . my bad

**Reply by cpurdy** [reply]
Hi Boardmad - what's your interest in Ecstasy? We'd be glad to offer some suggestions, and would welcome contributions :)

**Comment by Boardmad** [comment]
Hi Cameron
 We have a mutual friend (Phil Miller) who I’m working with at the moment. 👍
 Scott  
  
>

**Reply by cpurdy** [reply]
Excellent! Say hi to Phil for me :)

**Comment by Boardmad** [comment]
You bet!!👍   
  
>

**Comment by ggleyzer** [comment]
I don't see such a file. what are your "node" and "quasar" versions? I have:
    
    gui$ node --version
    v23.1.0
    gui$ quasar --version
    @quasar/app-vite 1.8.3 (@quasar/cli 2.3.0)
